framework/test/gtest/AnyMapTest.cpp

```diff
@@ -189,4 +195,21 @@ TEST(AnyMapTest, AnyMap)
   ASSERT_EQ(stream1.str(), "{do : 1, re : 2}");
   any_value_to_json(stream2, o_anymap1);
   ASSERT_EQ(stream2.str(), "{\"do\" : 1, \"re\" : 10}");
+}
+
+TEST(AnyMapTest, Move)
+{
+  AnyMap o(AnyMap::ORDERED_MAP);
+  o["do"] = Any(1);
+  o["re"] = Any(2);
+  AnyMap o_anymap_move_ctor(std::move(o));
+  ASSERT_EQ(any_cast<int>(o_anymap_move_ctor.at("do")), 1);
+  ASSERT_DEATH(o.size(), ".*")
+    << "This call should result in a crash because the object has been moved from";
+
+  AnyMap uo_anymap_move_assign(AnyMap::UNORDERED_MAP);
```

Comment:
I would split this into two test points. One test point for move_ctor another for move_assign